### DISCUSSION TITLE
Add elements for document sidebar and margin gutters

### DISCUSSION THREAD

**Discussion by jimjam-slam** [quarto-dev/quarto-cli#4659] [Feature Requests]
Quarto's CSS grid system has 13 grid column tracks:

![Screenshot of a Quarto document, with the grid lines highlighted by the browser tools](https://user-images.githubusercontent.com/6520659/223332412-385a1ec1-b1b4-44c6-badb-d2ac9e441a34.png)

The main elements in those tracks—the sidebar, article body (or header block) and margin, all have elements in the document, so they can be targeted by CSS.

![Main elements of the Quarto grid](https://quarto.org/docs/prerelease/1.3/images/grid.png)

The gutter elements, though (column tracks 1, 2, 6, 8, 12 and 13), are empty.

As far as I can tell, [the only way to target empty grid cells](https://stackoverflow.com/questions/46308048/how-to-target-a-specific-column-or-row-in-css-grid-layout) is with CSS Grid-Structural Selectors, which are still in development and don't yet have any browser support. Using `:nth-child` doesn't work, as that targets the nth HTML child, not the nth grid cell. In other words, it only works if every grid cell is occupied.

This isn't a big issue, but if the empty gutters had empty HTML elements in them, you could do things like have a sidebar or margin with a background colour that goes out to the page edge (say, if I wanted tracks 1–5 to all be blue).

**Comment by dragonstyle** [comment]
This is a pretty interesting suggestion - something I hadn't even thought of, so thank you! I'll open an enhancement request and give it some consideration for our 1.4 release (we'll starting working on that version soon, but are currently stabilizing 1.3). Two additional thoughts:

- Just wanted to be sure you were aware of the `docked` layout for the sidebar which may have behavior that is more to your liking.
- I wonder if we should be making the background behavior you describe just work if you side a sidebar background color (I have a feeling that the floating sidebar ends up with a band which probably doesn't look great).

Thanks again.

**Reply by jimjam-slam** [reply]
Oh yeah, the `docked` sidebar would probably a lot easier in most cases! I personally like the undocked sidebar sticking closer to the body at large widths, but that's very much to taste.

When you target `#quarto-sidebar`, you do indeed get a not great stripe:

<img width="1457" alt="image" src="https://user-images.githubusercontent.com/6520659/223883515-90f6fb2b-1a6f-45f1-86e5-711cc8eae457.png">

(Although if you were able to target the corresponding tracks in the `#title-block-header`, you'd be able to have a top-to-bottom stripe effect, which some people might like.)

You can also target `#TOC` for a floating pill effect, which is nice. See below with `background-color`, `padding-inset` and `border-radius`:

<img width="1457" alt="image" src="https://user-images.githubusercontent.com/6520659/223883985-4830b464-642c-4748-a820-20907caf7fb6.png">

This is all very much in enhancement territory. I do think there are some styling use cases you could enable by having elements in the otherwise empty grid cells (even if it is patching an unfortunate oversight in the grid system). For examplke, somebody might want to have a different background colour for the margin on the right. But I wouldn't call it high priority by any stretch!

**Comment by mcanouil** [comment]
Closing in favour of https://github.com/quarto-dev/quarto-cli/issues/4671
